src/mapblock_mesh.cpp

```diff
@@ -850,7 +850,13 @@ static void getTileInfo(
 			layer.material_flags |= MATERIAL_FLAG_BACKFACE_CULLING;
 	}
 
-	if (!data->m_smooth_lighting) {
+	bool fullbright = g_settings->getBool("fullbright_mode");
```

Comment:
This setting getter is too slow to use for every tile, fullbright mode needs to be cached in a bool.